### ISSUE TITLE
Confirm a function format to save and load incrementally

### ISSUE THREAD

**Issue by ellynliu** [uchicago-cs/chiventure#225]
Work with game state or action management to decide upon the incremental save and load function and how to call it, including deciding the name, parameters, return values.

On checkpointing side: We need to figure out how to get information from game state and serialize it (which may involve creating a new struct) and read from the file and load it.

On game state or action management side: You need to figure out how to call our functions whenever your struct changes.

**Comment by jarvislam1999** [comment]
After talking with game-state and Mark, we have decided on the skeleton and rough format of the functions as follow:

We will likely build a checkpointing_log_TYPE() function to call on different changed members of structs. For example, when we want to change a player's xp, which is an integer, we can call:
checkpointing_log_int("PLAYER", "playerid.xpid", old_int_value, new_int_value)
inside whichever function game-state uses to change a player's xp (maybe set_xp_to() or something similar?) 
where the first argument is probably the struct you will call it in (ROOM, PLAYER, ITEM, GAME, ...), the second one is probably the id of the player followed by which member changed (so like player1.xp to change the xp of player1), the third and fourth values are the intiget values that player1->xp stores.

The function format will likely be more refined over time, but I just want to give game-state and potentially other teams a head-up. I probably won't start this until we have a working prototype.

**Comment by jarvislam1999** [comment]
Mark and I had a conversation where we agreed that with the time remaining, we should not try to implement incremental game save. However, we reached a compromise where I will be writing the preliminary save functions that game-state is supposed to call (we will not integrate that into game state for this sprint so you guys won't actually have to include it in your functions). We should communicate and agree on this function format, a.k.a whether the format can be called in game state and then I can just write them. @lindsey-hsi What do you think about the function formats above?

**Comment by jarvislam1999** [comment]
Instead of the current format checkpointing_log_int("PLAYER", "playerid.xpid", old_int_value, new_int_value), we should have checkpointing_log_int("PLAYER", "playerid", "xpid", old_int_value, new_int_value) where the two ids are split for convenience purposes.

**Comment by jarvislam1999** [comment]
I have finished writing the function and the final format is at follow:

/* Store a changed value information in our array of logs
 * Parameters:
 * - logs: the array of logs
 * - level: either "game", "player", "room", "item", etc
 * - player_id : the id of that level string
 * - member : member of the level string
 * - old_value: old value of that member
 * - new_value: new value of that member
 * Return: 0 if SUCCESS, -1 OTHERWISE
 */
int checkpointing_log_int(Logs *logs, char *level, char *level_id, char *member\
, int old_value, int new_value);

I added logs as a parameter so that game state can add logs into it. logs can be easily intialized using logs_new() explained in log_save.h

Within this sprint, there's no need for game state to begin using this, however. They are expected to use this whenever they change a value to the game_t struct. For example, the health of player would be called:
checkpointing_log_int("player", "chad", "health", 100, 0);
when chad's health changed from 100 to 0

If game state is fine with this, this issue can then be closed.
